data/src/main/java/me/androidbox/data/alarm_manager/AlarmReceiver.kt

```diff
@@ -1,15 +1,75 @@
 package me.androidbox.data.alarm_manager
 
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
 import android.content.BroadcastReceiver
 import android.content.Context
 import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import me.androidbox.data.R
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_AGENDA_TYPE
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_DESCRIPTION
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_ID
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_TITLE
 
 class AlarmReceiver : BroadcastReceiver() {
 
     override fun onReceive(context: Context?, intent: Intent?) {
-        intent?.getStringExtra("EXTRA_MESSAGE")?.let { message ->
-            /* TODO Add Notification here */
-            println("Alarm triggered: $message")
+        /* TODO Not sure if intent and context can be null but I guess is ok to cast it anyway */
+        if(intent != null && context != null) {
+            intent.extras?.let { bundle ->
+                val agendaId = bundle.getString(EXTRA_ID) ?: ""
+                val title = bundle.getString(EXTRA_TITLE) ?: "Title"
+                val description = bundle.getString(EXTRA_DESCRIPTION) ?: "Description"
+                val agendaType = bundle.getString(EXTRA_AGENDA_TYPE) ?: "Agenda"
+
+                if (agendaId.isNotBlank()) {
+                    createNotificationChannel(context, agendaId, agendaType)
+                    createAndShowNotification(context, agendaId, title, description)
+                }
+            }
+        }
+    }
+
+    private fun createNotificationChannel(context: Context, agendaId: String, agendaName: String) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationManager = getNotificationManager(context)
+            val notificationChannel = NotificationChannel(
+                agendaId,
+                agendaName,
+                NotificationManager.IMPORTANCE_HIGH
+            )
+
+            notificationManager.createNotificationChannel(notificationChannel)
         }
```

Comment:
@MKiperszmid @philipplackner Is it ok to use the agendaId for the channelId, which could be the eventId, taskId, or reminderId for what alarm is be scheduled for?

Reply:
Nono this is bad
Since you'll create a new channel for each agenda item, and that's a big no-no

You could have either
1 channel to display all agendaitem notifications
Or 3 channels, one for each type of notification

Reply:
@MKiperszmid 
I have used the following for the channel name, which will appear in the app settings on the device:
`val channelName = "Notifications for $agendaType"`

And for the channel ID I am using the `agendaType` (event, task, reminder)

So would this create a channel for notifications for each agenda type, event, task, or reminder

Would this be correct now?

Reply:
Yea! That´s good

Reply:
Yes, one channel per notification type is great